File: 백준/Bronze/14405. 피카츄/피카츄.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.next();
        boolean flag = false;

        for (int i = 0; i < s.length(); i++) {
            if (i < s.length() - 1 && (s.substring(i, i + 2).equals("pi") || s.substring(i, i + 2).equals("ka"))) {
                i += 1;
            } else if (i < s.length() - 2 && s.substring(i, i + 3).equals("chu")) {
                i += 2;
            } else {
                flag = true;
            }
        }

        if (flag) {
            System.out.println("NO");
        } else {
            System.out.println("YES");
        }
    }
}

File: 백준/Silver/14405. 피카츄/피카츄.py
s = input().strip()
i = 0
is_good = True

while i < len(s):
    if i + 2 <= len(s) and (s[i:i+2] == 'pi' or s[i:i+2] == 'ka'):
        i += 2
    elif i + 3 <= len(s) and s[i:i+3] == 'chu':
        i += 3
    else:
        is_good = False
        break

# 문자열 끝까지 도달했는지 확인
if is_good and i == len(s):
    print("YES")
else:
    print("NO")